include/rocksdb/public_port.h

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2011 The LevelDB Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+#pragma once
+
+// Best used after all the headers in case windows #defines give you trouble
+// Minimum subset
+
+#ifndef ROCKSDB_PUBLIC_PORT_H__
+#define ROCKSDB_PUBLIC_PORT_H__
+
+#ifdef OS_WIN
+
+#include <stdint.h>
+
+#undef min
+#undef max
+#undef DeleteFile
+#undef GetCurrentTime
+
+#ifndef strcasecmp
+#define strcasecmp _stricmp
+#endif
+
+#ifndef snprintf
+#define snprintf _snprintf
+#endif
+
+#ifndef ROCKSDB_PRIszt
+#define ROCKSDB_PRIszt "Iu"
+#endif
```

Comment:
Db.h, env.h, perf_context.h and some others. With .cc files it is easy to include port/port.h

Dmitri

From: Siying Dong [mailto:notifications@github.com]
Sent: Monday, July 13, 2015 5:53 PM
To: facebook/rocksdb
Cc: Dmitri Smirnov
Subject: Re: [rocksdb] Ensure Windows build w/o port/port.h in public headers (#652)


In include/rocksdb/public_port.h<https://github.com/facebook/rocksdb/pull/652#discussion_r34528738>:

> +#undef min

> +#undef max

> +#undef DeleteFile

> +#undef GetCurrentTime

> +

> +#ifndef strcasecmp

> +#define strcasecmp _stricmp

> +#endif

> +

> +#ifndef snprintf

> +#define snprintf _snprintf

> +#endif

> +

> +#ifndef ROCKSDB_PRIszt

> +#define ROCKSDB_PRIszt "Iu"

> +#endif

Which public .h file depends on them? Can we simply move it to .cc files?

—
Reply to this email directly or view it on GitHub<https://github.com/facebook/rocksdb/pull/652/files#r34528738>.